src/dune_pkg/local_package.ml

```diff
@@ -145,7 +145,26 @@ let of_package (t : Dune_lang.Package.t) =
   let name = Package.name t in
   match Package.original_opam_file t with
   | None ->
-    let dependencies = t |> Package.depends |> Dependency_formula.of_dependencies in
+    let depends =
+      match Package.depends t, Package.has_opam_file t with
+      | [], Exists true ->
```

Comment:
:question: This means that we don't do any distinction between `(package name)` and `(package name (depends))`?

Reply:
We cannot, `Package.t` does not distinguish between empty and unspecified for `depends`.

In #11555 we discussed that there is no purpose in allowing to load data from multiple sources so if `(package (name foo))` is declared, that means that this is the canonical source for the `Package.t` and this package has no dependencies. So there is no semantic difference between specifying an empty `(depends)` vs no depends at all and we might as well continue representing it as `[]`.

Reply:
Make sense, thanks for the clarification!